### PR TITLE
docs(color-picker): improve example

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    --       |
| 🇨🇳 Chinese |     改进组件示例      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 4c9f757</samp>

Refactored the color picker demo files to use reusable components, custom hooks, and dynamic presets. Renamed some files to match the format prop of the color picker. Simplified the code and improved the performance.

### <samp>🤖 Generated by Copilot at 4c9f757</samp>

*  Refactor color picker demo to use different formats and custom panel layouts ([link](https://github.com/ant-design/ant-design/pull/45527/files?diff=unified&w=0#diff-7be4f6ff081124d854515c9e6f08a451f4073636c9526bd71c1cb3408f0c6802L2-R7),[link](https://github.com/ant-design/ant-design/pull/45527/files?diff=unified&w=0#diff-7be4f6ff081124d854515c9e6f08a451f4073636c9526bd71c1cb3408f0c6802R14-R30),[link](https://github.com/ant-design/ant-design/pull/45527/files?diff=unified&w=0#diff-7be4f6ff081124d854515c9e6f08a451f4073636c9526bd71c1cb3408f0c6802R36-R52),[link](https://github.com/ant-design/ant-design/pull/45527/files?diff=unified&w=0#diff-7be4f6ff081124d854515c9e6f08a451f4073636c9526bd71c1cb3408f0c6802L29-R77),[link](https://github.com/ant-design/ant-design/pull/45527/files?diff=unified&w=0#diff-e00c41c079465e93600695586a06b1a5777aad2bf2ca30ea7f97c7a953a58550L2-R102),[link](https://github.com/ant-design/ant-design/pull/45527/files?diff=unified&w=0#diff-223d7ae6b7c1e189b58f0a96c353e9a5016878eb74b7675fff666216f9bd828dL2-R36))
  - Remove unused imports of `Col`, `Row`, and `Space` from `format.tsx` ([link](https://github.com/ant-design/ant-design/pull/45527/files?diff=unified&w=0#diff-7be4f6ff081124d854515c9e6f08a451f4073636c9526bd71c1cb3408f0c6802L2-R7))
  - Define `HexCase`, `HsbCase`, and `RgbCase` components to render color picker with hex, hsb, and rgb values respectively ([link](https://github.com/ant-design/ant-design/pull/45527/files?diff=unified&w=0#diff-7be4f6ff081124d854515c9e6f08a451f4073636c9526bd71c1cb3408f0c6802R14-R30),[link](https://github.com/ant-design/ant-design/pull/45527/files?diff=unified&w=0#diff-7be4f6ff081124d854515c9e6f08a451f4073636c9526bd71c1cb3408f0c6802R36-R52),[link](https://github.com/ant-design/ant-design/pull/45527/files?diff=unified&w=0#diff-7be4f6ff081124d854515c9e6f08a451f4073636c9526bd71c1cb3408f0c6802L29-R77))
  - Simplify default export function to return a `Space` component with the three cases as children ([link](https://github.com/ant-design/ant-design/pull/45527/files?diff=unified&w=0#diff-7be4f6ff081124d854515c9e6f08a451f4073636c9526bd71c1cb3408f0c6802L29-R77))
  - Rename `panel-render.tsx` to `format.tsx` and use `HorizontalLayoutDemo` and `BasicDemo` components to render color picker with custom panel layouts ([link](https://github.com/ant-design/ant-design/pull/45527/files?diff=unified&w=0#diff-e00c41c079465e93600695586a06b1a5777aad2bf2ca30ea7f97c7a953a58550L2-R102))
  - Rename `presets.tsx` to `format.tsx` and use `genPresets` function to generate presets from theme token and color palettes ([link](https://github.com/ant-design/ant-design/pull/45527/files?diff=unified&w=0#diff-223d7ae6b7c1e189b58f0a96c353e9a5016878eb74b7675fff666216f9bd828dL2-R36))

### CODE REVIEW

File: components/color-picker/demo/panel-render.tsx [hunk 1]
```diff
@@ -1,125 +1,102 @@
...
+      <Col flex="auto">
+        <Picker />
+      </Col>
     </Row>
-  </Space>
+  );
+
+  return (
+    <ColorPicker
+      open
```
Comment (Wxh16144) [inline components/color-picker/demo/panel-render.tsx:62]:
```suggestion
```

File: components/color-picker/demo/presets.tsx [hunk 2]
```diff
@@ -1,48 +1,36 @@
...
+    label: upperFirst(label),
+    colors: [
+      ...colors,
+      ...colors.map(
+        // Add transparency
+        (color, idx) => new TinyColor(color).setAlpha(idx / 10).toHex8String(),
+      ),
+    ],
+  }));
+
```
Comment (RedJue) [inline components/color-picker/demo/presets.tsx:21]: 对 **demo** 而言写法感觉有点过于复杂了，尽量不用第三方库来实现，还是以简单直观为主

File: components/color-picker/ColorPickerPanel.tsx [hunk 3]
```diff
@@ -45,18 +46,25 @@ const ColorPickerPanel: FC<ColorPickerPanelProps> = (props) => {
...
-            ? panelRender(innerPanel, {
-                components: {
-                  Picker: PanelPicker,
-                  Presets: PanelPresets,
-                },
-              })
-            : innerPanel}
+        <div
+          className={classNames(colorPickerPanelPrefixCls, {
+            [`${colorPickerPanelPrefixCls}-customized`]: panelNode !== innerPanel,
```
Comment (MadCcc) [inline components/color-picker/ColorPickerPanel.tsx:64]: 用这个 class 去做筛选不太好，默认的宽度还是要给

File: components/color-picker/demo/presets.tsx (hunk 2 above)
Comment (MadCcc) [reply components/color-picker/demo/presets.tsx]: +1，demo 尽量通俗易懂

Comment (Wxh16144) [reply components/color-picker/demo/presets.tsx]: 嗯嗯，简化了一些，demo 还是保留了 `@ant-design/colors`，期望开发者使用以符合 [Ant Design 色彩体系](https://ant.design/docs/spec/colors-cn#%E5%9C%A8%E4%BB%A3%E7%A0%81%E4%B8%AD%E4%BD%BF%E7%94%A8%E8%89%B2%E6%9D%BF)

File: components/color-picker/ColorPickerPanel.tsx (hunk 3 above)
Comment (Wxh16144) [reply components/color-picker/ColorPickerPanel.tsx:64]: 有什么建议吗🤔

**Comment by afc163** [comment]
图片

这一块好好写写，有助于了解这个 PR 的背景和解决思路。

**Comment by Wxh16144** [comment]
> 这一块好好写写，有助于了解这个 PR 的背景和解决思路。

嗯嗯，等最后确定解决方案后，我会补充的

File: components/color-picker/__tests__/__snapshots__/index.test.tsx.snap [hunk 4]
```diff
@@ -50,7 +50,7 @@ exports[`ColorPicker Should panelRender work 1`] = `
           class="ant-popover-inner-content"
         >
           <div
-            class="ant-color-picker-inner-content"
+            class="ant-color-picker-inner-content ant-color-picker-inner-content-customized"
```
Comment (afc163) [inline components/color-picker/__tests__/__snapshots__/index.test.tsx.snap:53]: 这个命名不要，印象里 antd 里没有因为增加了自定义 render，就去标记某个节点被 `自定义过` 的先例。

File: components/color-picker/style/index.ts [hunk 5]
```diff
@@ -179,17 +179,22 @@ const genColorPickerStyle: GenerateStyle<ColorPickerToken> = (token) => {
...
 
+  const innerContent = `${componentCls}-inner-content`;
+
   return [
     {
       [componentCls]: {
-        [`${componentCls}-inner-content`]: {
-          display: 'flex',
-          flexDirection: 'column',
-          width: colorPickerWidth,
```
Comment (afc163) [inline components/color-picker/style/index.ts:188]: 我试了一下，可以把原来的这三行预设代码删掉，替换成一句`minWidth: colorPickerWidth`（这居然不影响现在的布局）。

然后去掉 customized 相关 className 和样式即可。这样 panelRender 里的 flex 布局就不会被外部影响了。

Comment (Wxh16144) [reply components/color-picker/style/index.ts:188]: 首要前提是不影响现在布局，然后做减法。

设置最小宽度后，自定义的时候也没办法自由宽度了呀

Comment (afc163) [reply components/color-picker/style/index.ts:188]: 可以把宽度放 `.ant-color-picker-panel` 和 `.ant-color-picker-presets` 上，用内容去顶开容器宽度，容器上不预设宽度。

Comment (afc163) [reply components/color-picker/style/index.ts:188]: 我感觉 minWidth 放在 inner-content 问题不是很大，这个 minWidth 可以被 ConfigProvider 自定义就行。

Comment (Wxh16144) [reply components/color-picker/style/index.ts:188]: > 我感觉 minWidth 放在 inner-content 上问题不是很大，这个 minWidth 可以被 ConfigProvider 或 styles 属性自定义就行。

这样其实也还行~ 另外一个解法是直接再套个 div

image

Comment (Wxh16144) [reply components/color-picker/style/index.ts:188]: > 这个 minWidth 可以被 ConfigProvider 自定义就行。
现在整个 color-picker 都不支持 CP components 自定义 token

Comment (Wxh16144) [reply components/color-picker/style/index.ts:188]: 这里的逻辑请移步至：https://github.com/ant-design/ant-design/pull/46327  讨论，CC @afc163 @MadCcc @RedJue

Comment (Wxh16144) [reply components/color-picker/style/index.ts:188]: 这个 PR 之后仅做 demo 的更新， demo 使用 @ant-design/colors，期望开发者使用以符合 [Ant Design 色彩体系](https://ant.design/docs/spec/colors-cn#%E5%9C%A8%E4%BB%A3%E7%A0%81%E4%B8%AD%E4%BD%BF%E7%94%A8%E8%89%B2%E6%9D%BF)

**Review by zombieJ — APPROVED** [review]

**Review by MadCcc — APPROVED** [review]

**Pull request merged**